project/frontend/src/components/PreferenceForm.js

```diff
@@ -0,0 +1,87 @@
+import './PreferenceForm.css';
+
+import React from 'react';
+
+import CuisineAutocomplete from './CuisineAutocomplete.js';
+import FormItem from './FormItem.js';
+
+import Place from '../assets/place.svg';
+import Cuisine from '../assets/cuisine.svg';
+import Distance from '../assets/distance.svg';
+import Experience from '../assets/food_service.svg';
+import Price from '../assets/dollar.svg';
+
+/**
+ * A form used to help gather user preferences for the 'Find My Match' and Profile Page.
+ *
+ * @param {!Object<string, string>} rowItemLabels The labels for each form input item.
+ * @param {boolean} isUserPreference  True if the use of this form is for the user
+ *      preference form on the home page.
+ * @param {string} locationName The formatted address of the current user.
+ * @param {string} buttonLabel The label on the submit button.
+ * @param {function} handleSubmit The function called when the submit button is pressed.
+ * @param {!Array<string>} cuisineOptions The list of cuisines that will be shown
+ *      in the cuisine autocomplete component.
+ * @param {function} setCuisine A cuisine setter function that will update
+ *      when something is added to autocomplete.
+ */
+function PreferenceForm(props) {
+  const {
+    rowItemLabels,
+    isUserPreference,
+    locationName,
+    buttonLabel,
+    handleSubmit,
+    cuisineOptions,
+    setCuisine,
+  } = props;
+  return (
+    <div className='form-container'>
+      <form className='form' onSubmit={handleSubmit}>
+        <FormItem imageName={Place} label={rowItemLabels.location}>
+          <p>{locationName}</p>
+        </FormItem>
+        <FormItem imageName={Cuisine} label={rowItemLabels.cuisine}>
+          <CuisineAutocomplete
+            cuisineOptions={cuisineOptions}
+            setCuisine={setCuisine}
+          />
+        </FormItem>
+        <div className='form-row'>
+          <div className='form-column'>
+            <FormItem imageName={Distance} label={rowItemLabels.distance}>
+              {props.children[0]}
```

Comment:
What are these specific indeces from props.children/is there a way to make it less hard codey to prevent index out of bounds?

Reply:
Unfortunately no (for now). I've discussed alternatives with Jesse but this seems to be the best way for now. Luckily an out of bounds error wont happen when the user side just on the developer side if someone were to change props.children. Ive added documentation accounting for this since it isn't ideal:
https://github.com/googleinterns/step127-2020/blob/cd2125a364c3b020d0609df7e2127dfb0a637e19/project/frontend/src/components/ProfileForm.js#L36